Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -100,8 +107,76 @@ public bool SetCursorPosition (Coord position)
 			return SetConsoleCursorPosition (ScreenBuffer, position);
 		}
 
+		public void SetInitialCursorVisibility ()
+		{
+			if (initialCursorVisibility.HasValue == false && GetCursorVisibility (out CursorVisibility visibility)) {
+				initialCursorVisibility = visibility;
+			}
+		}
+
+		public bool GetCursorVisibility (out CursorVisibility visibility)
+		{
+			if (!GetConsoleCursorInfo (ScreenBuffer, out ConsoleCursorInfo info)) {
+
+				var err = Marshal.GetLastWin32Error ();
+
+				if (err != 0) {
+					throw new System.ComponentModel.Win32Exception (err);
+				}
+
+				visibility = Gui.CursorVisibility.Default;
+
+				return false;
+			}
+
+			if (!info.bVisible)        visibility = CursorVisibility.Invisible;
+			else if (info.dwSize > 50) visibility = CursorVisibility.Box;
+			else                       visibility = CursorVisibility.Underline;
+
```

Comment:
I see what is going on here, but let us stick to the coding style we use here, each`visibility = XX` needs to go on its own line.